src/main/java/gov/usds/case_issues/controllers/UserInformationApiController.java

```diff
@@ -1,38 +1,34 @@
 package gov.usds.case_issues.controllers;
 
-import java.util.HashMap;
-
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.context.annotation.Profile;
 import org.springframework.http.HttpStatus;
 import org.springframework.http.ResponseEntity;
 import org.springframework.security.core.Authentication;
 import org.springframework.web.bind.annotation.GetMapping;
 import org.springframework.web.bind.annotation.RestController;
 
+import gov.usds.case_issues.model.SerializedUserInformation;
+import gov.usds.case_issues.services.UserService;
+
 /**
  * Controller to see information about the current user.
  */
 @RestController
 @Profile("auth-testing")
-public class UserDetailsApiController {
+public class UserInformationApiController {
+
+	@Autowired
+	private UserService _userService;
 
 	public static final String USER_INFO_ENDPOINT = "/api/users";
 
-	@GetMapping(UserDetailsApiController.USER_INFO_ENDPOINT)
-	public Object getUser(Authentication u) {
-		HashMap<String, String> user = new HashMap<String, String>();
-		String name = u.getName();
-		if (name.contains(";")) {
-			String[] userInfo = name.split(";");
-			user.put("ID", userInfo[0]);
-			user.put("name", userInfo[1]);
-		} else {
-			user.put("name", u.getName());
-		}
-		return user;
+	@GetMapping(UserInformationApiController.USER_INFO_ENDPOINT)
+	public SerializedUserInformation getCurrentUser(Authentication auth) {
+		return _userService.getCurrentUser(auth);
 	}
 
-	@GetMapping(UserDetailsApiController.USER_INFO_ENDPOINT + "/loggedin")
+	@GetMapping(UserInformationApiController.USER_INFO_ENDPOINT + "/loggedin")
```

Comment:
Is this actually used?

Reply:
I don't think it is. I will submit another PR to remove it